src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -41,50 +42,87 @@ const SelectPanelContext = React.createContext<{
   selectionVariant: 'multiple',
 })
 
-// @ts-ignore todo
-const SelectPanel = props => {
-  const anchorRef = useProvidedRefOrCreate(props.anchorRef)
+export type SelectPanelProps = {
+  title: string
+  description?: string
+  selectionVariant?: ActionListProps['selectionVariant'] | 'instant'
+  id?: string
+
+  defaultOpen?: boolean
+  open?: boolean
+  anchorRef?: React.RefObject<HTMLButtonElement>
+
+  onCancel?: () => void
```

Comment:
Do you think we need to type the event argument here? Something like this or not needed?

```
event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement>
```

Reply:
There are multiple elements and events to cancel: click/press cancel if it's visible, click/press the cross icon (always present), click outside. So, I'm making `onCancel` a callback for all of these without passing event. 

I don't think there is a use case yet to access the event or prevent it, let's see if we get feedback based on usage around this!